src/api/cpp/telemetry/telemetry_plugin.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _NIXL_SRC_API_CPP_TELEMETRY_TELEMETRY_PLUGIN_H
+#define _NIXL_SRC_API_CPP_TELEMETRY_TELEMETRY_PLUGIN_H
+
+#include "telemetry/telemetry_exporter.h"
+#include <string_view>
+#include <memory>
+
+enum class NixlTelemetryPluginApiVersion : unsigned int { V1 = 1 };
+
+inline constexpr NixlTelemetryPluginApiVersion NIXL_TELEMETRY_PLUGIN_API_VERSION =
+    NixlTelemetryPluginApiVersion::V1;
+
+// Type alias for exporter creation function
+using ExporterCreatorFn =
```

Comment:
```suggestion
using exporter_creator_fn_t =
```
https://github.com/ai-dynamo/nixl/blob/main/docs/CodeStyle.md